addons/mail/static/src/core/common/thread_model.js

```diff
@@ -975,20 +984,10 @@ export class Thread extends Record {
         ) {
             if (this.channel_type === "channel" || this.channel_type === "group") {
                 this.name = newName;
-                await this.store.env.services.orm.call(
-                    "discuss.channel",
-                    "channel_rename",
-                    [[this.id]],
-                    { name: newName }
-                );
+                await this.renameChannel(newName);
             } else if (this.channel_type === "chat") {
                 this.custom_channel_name = newName;
-                await this.store.env.services.orm.call(
-                    "discuss.channel",
-                    "channel_set_custom_name",
-                    [[this.id]],
-                    { name: newName }
-                );
+                await this.renameChannel(newName, "channel_set_custom_name");
```

Comment:
This branching should be invoked with live chat too, rather than copying its code in `renameChannel` with special parameter.
This would reuse the same code for `custom_channel_name` for both `chat` and `live chat` as desired, with minimal code maintenance.